api/src/pcapi/routes/backoffice/offers/blueprint.py

```diff
@@ -525,8 +525,8 @@ def batch_edit_offer() -> utils.BackofficeResponse:
     offers = offers_models.Offer.query.filter(offers_models.Offer.id.in_(form.object_ids_list)).all()
     criteria = criteria_models.Criterion.query.filter(criteria_models.Criterion.id.in_(form.criteria.data)).all()
 
-    previous_criteria = list(set.intersection(*[set(offer.criteria) for offer in offers]))
-    deleted_criteria = list(set(previous_criteria).difference(criteria))
+    previous_criteria = set.intersection(*[set(offer.criteria) for offer in offers])
+    deleted_criteria = list(previous_criteria.difference(criteria))
```

Comment:
Il me semble que `deleted_criteria` n'a pas besoin d'être une liste. On ne fait qu'itérer dessus quelques lignes plus bas, ça marchera tout aussi bien avec un set ! :) 

```suggestion
    deleted_criteria = previous_criteria.difference(criteria)
```